### PR TITLE
New: CU-2ub0937 Added error message when a registration step is not OK

### LINKED ISSUE #2274: Password hint is not visible on Android
During registration, when a password is not strong enough we show a warning to users and give them a hint of what their password must include.
On Android, we show this in the Toast element. This, however, is not capable of showing such a long message (at least from >Android 12),. 
As a result, users do not really know what are the password constraints.

Moreover, on Herman's phone it only said "Password is not strong enough.".

<table>
<tr>
<th>
iOS
</th>
<th>
Android
</th>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
</table>

### PR DESCRIPTION
**Pull request by iiLubos**
Updated style of the error messages in the registration form

Fixes #2274

### CODE REVIEW

File: core/merginapi.cpp [hunk 1]
```diff
@@ -759,49 +759,55 @@ void MerginApi::registerUser( const QString &username,
...
+
   }
 
   if ( confirmPassword != password )
   {
-    emit registrationFailed();
-    emit notify( tr( "Passwords do not match" ) );
+    QString msg = tr( "Passwords do not match" );
+    emit registrationFailed( msg, RegistrationError::RegistrationErrorType::CONFIRM_PASSWORD );
+    emit notify( msg );
```
Comment (PeterPetrik) [inline core/merginapi.cpp:802]: if we have red error text below the offending field, we do not want to show toast at all I believe

**Comment by PeterPetrik** [comment]
@iiLubos do you have some screenshot of new GUI ?

**Review by tomasMizera — COMMENTED** [review]
- preview 
- tests
- commented out code blocks

**Comment by tomasMizera** [comment]
Note: toasts on Android are changing to popups similiar to iOS/desktop popups

File: app/qml/RegistrationForm.qml [hunk 2]
```diff
@@ -34,6 +34,48 @@ Rectangle {
     acceptTOC.checked = false
   }
 
+  Connections {
+    target: __merginApi
+    enabled: registerForm.visible
+
+    // show error message under the respective field
+    function onRegistrationFailed( msg, field ) {
+      console.log( "onRegistrationFailed: " + msg )
```
Comment (tomasMizera) [inline app/qml/RegistrationForm.qml:43]: Let's remove this debug log

**Review by tomasMizera — APPROVED** [review]
The code is looking good, I can not build it locally as I am still on the 6.4.2, so let's forward it to testing :)

**Comment by PeterPetrik** [comment]
autotests are repaired in my PR, fail unrelated to this PR

**Pull request merged**